src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs

```diff
@@ -199,16 +199,10 @@ private string GetResponseContent(HttpResponseMessage response, Guid requestGuid
         {
             try
             {
-                var responseStream = response.Content.ReadAsStreamAsync().GetAwaiter().GetResult();
+                var responseStream = response.Content?.ReadAsStreamAsync().GetAwaiter().GetResult();
 
                 if (responseStream == null)
-                {
-                    throw new NullReferenceException("responseStream");
-                }
-                if (response.Headers == null)
```

Comment:
Do we still need to be defensive against this scenario?

Reply:
I don't think so, the underlying implementation guarantees the property isn't null. https://github.com/microsoft/referencesource/blob/master/System/net/System/Net/Http/HttpResponseMessage.cs#L94